staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -69,50 +70,69 @@ func DefaultStacktracePred(status int) bool {
 	return (status < http.StatusOK || status >= http.StatusInternalServerError) && status != http.StatusSwitchingProtocols
 }
 
-// NewLogged turns a normal response writer into a logged response writer.
+// WithLogging wraps the handler with logging.
+func WithLogging(handler http.HandlerFunc, pred StacktracePred) http.HandlerFunc {
```

Comment:
Take and return `Handler` not `HandlerFunc`

```go
func WithLogging(handler http.Handler, pred StacktracePred) http.Handler {
```

Reply:
done